internal/analytics/default_reporter.go

```diff
@@ -0,0 +1,197 @@
+package analytics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/kubeshop/botkube/pkg/version"
+	"gopkg.in/yaml.v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/strings"
+
+	segment "github.com/segmentio/analytics-go"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	kubeSystemNSName     = "kube-system"
+	analyticsFileName    = "analytics.yaml"
+	printAPIKeyCharCount = 3
+)
+
+var (
+	// APIKey contains the API key for external analytics service. It is set during application build.
+	APIKey string
+)
+
+var _ Reporter = &DefaultReporter{}
+
+// DefaultReporter is a default Reporter implementation.
+type DefaultReporter struct {
```

Comment:
```suggestion
type SegmentReporter struct {
```

in that way we know what it really does.